roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -85,10 +87,27 @@ impl ParseClientJobDeclarationMessages for JobDeclaratorDownstream {
 
             for (i, sid) in short_hash_list.iter().enumerate() {
                 let sid_: [u8; 6] = sid.to_vec().try_into().unwrap();
-                if let Some(tx_data) = short_id_mempool.get(&sid_) {
-                    txs_in_job.push(tx_data.clone());
-                } else {
-                    missing_txs.push(i as u16);
+                match short_id_mempool.get(&sid_) {
+                    Some(tx_data) => match &tx_data.tx {
+                        Some(tx) => txs_in_job.push(tx.clone()),
+                        None => {
+                            let new_tx_data = self
+                                .mempool
+                                .safe_lock(|x| x.get_client())
+                                .unwrap()
+                                .unwrap()
+                                .get_raw_transaction(&tx_data.id.to_string(), None);
+                            if let Ok(new_tx) = new_tx_data {
```

Comment:
in the case that `new_tx_data` is `Err(...)` should we add the index to the list of missing transactions? like below `missing_txs.push(i as u16),`